src/lib/Polykey.ts

```diff
@@ -17,21 +18,22 @@ class Polykey {
     polykeyPath: string = `${os.homedir()}/.polykey`,
     fileSystem: typeof fs,
     keyManager?: KeyManager,
-    vaultManager?: VaultManager,
     peerManager?: PeerManager,
+    vaultManager?: VaultManager,
   ) {
```

Comment:
I think if you are going to have separate objects for `Agent` and `Client` it actually doesn't make sense to have a `Polykey` class at all. Like what does it even represent? Or perhaps...

You need a:

* A class to represent the lifecycle of PK `Client` (which is constructed by the CLI commands), but also potentially by other users to js-polykey (like Polykey GUI)
* A class to represent the lifecycle of PK `Agent`
* A class to represent the connection to the Agent from the Client. It's as if the connection to Agent is really a property inside the Client. And furthermore the connection to Agent is not `Agent`. Just like how a database connection is not the database. Therefore you might want to differentiate this and call it `AgentConnection`.
* Separate models to encapsulate the different kinds of actions and state you want to manipulate. So `Agent` will be using these models directly in-memory. While `Client` will be performing these actions over the network. this means `Agent` uses core models. `Client` models are proxy models. However `Client` models should not be a carbon copy with just wrappers to the `Agent` models. The `Agent`  models are lower level. For example, `Agent` understands git repos, and so my have a model for git ops and git state. But `Client` only understands vaults, and so it may have a vault based model. That being said `Agent` may still have a `Vault` model. But the `Client`'s `Vault` model is just performing RPC calls and really high level ones too, since `Client` does not have direct access to the actual state.

Can we do some UML drafting here to make this really clear. Better put it into an issue.